akka-cluster/src/main/scala/akka/cluster/ClusterSettings.scala

```diff
@@ -117,13 +117,32 @@ final class ClusterSettings(val config: Config, val systemName: String) {
   val AllowWeaklyUpMembers = cc.getBoolean("allow-weakly-up-members")
 
   val DataCenter: DataCenter = cc.getString("data-center")
+
+  trait CrossDcFailureDetectorSettings {
+    protected def multiDcConfig: Config
+    lazy val ImplementationClass: String = multiDcConfig.getString("implementation-class")
```

Comment:
what is the purpose of the trait? the `lazy` is not so nice, doesn't matter here but might be a non-optimal pattern for other places

I'd prefer an ordinary `final class` and
`val CrossDcFailureDetectorSettings = new